Fit to rubocop rules

huyu398 opened the pull request:
コード保守のため， rubocop を導入し，ソースコードの修正を行いました．

huyu398 committed:
Configure rubocop rules for `dydx` lib.

huyu398 committed:
Modify `dydx` lib with rubocop rules.

huyu398 commented:
まだ， rspec のコード修正が出来ていないので， WIP です．

gogotanaka commented:
すみませんファイル末改行やスペースなし=などは完全に僕の怠惰によるものです...

回収して頂き大変嬉しいです！

ps. 先ほどの僕のcommit でコンフリクトしたっぽいです、、！すみません、、、！

よしなに変更しといて下さいませ

huyu398 commented:
@gogotanaka 
おそらく，積分メソッドあたりでしょう……
rspec の方も修正し終わったら，競合解消して commit したいと思います!

gogotanaka commented:
@huyu398 すみません、

僕が投げているブランチでSet モジュールに対する大きめのリファクタをしてしまったので、

かなりコンフリクト起きるかもです。。

https://github.com/gogotanaka/dydx/commit/fe63933f61ec653e8526f417371b2df76b36e04b

もしあれでしたら、こちらを先に入れて頂いて、僕の方で競合解消しますがどうでしょうか

ps. すみません、この変更早めに入れたかったので入れてしまいました。。

かなりコンフリクトが起きると思われますが、よしなにお願い致します。

ご迷惑おかけ致します。。。

huyu398 commented:
@gogotanaka 
了解しました!

huyu398 committed:
Configure rubocop rules for `dydx` spec.

huyu398 committed:
Modify `dydx` spce with rubocop rules.

huyu398 committed:
Modify Rakefile and dydx.gemspec with rubocop rules.

huyu398 committed:
Merge remote-tracking branch 'upstream/master' into fit_rubocop

Conflicts:
	lib/dydx/algebra.rb
	lib/dydx/algebra/inverse.rb
	lib/dydx/algebra/operator.rb
	lib/dydx/algebra/set/base.rb
	lib/dydx/algebra/set/cos.rb
	lib/dydx/algebra/set/e.rb
	lib/dydx/algebra/set/fixnum.rb
	lib/dydx/algebra/set/float.rb
	lib/dydx/algebra/set/log.rb
	lib/dydx/algebra/set/num.rb
	lib/dydx/algebra/set/pi.rb
	lib/dydx/algebra/set/sin.rb
	lib/dydx/algebra/set/symbol.rb
	lib/dydx/algebra/set/tan.rb
	lib/dydx/integrand.rb
	lib/dydx/version.rb
	spec/dydx_spec.rb
	spec/lib/algebra/set/cos_spec.rb
	spec/lib/algebra/set/e_spec.rb
	spec/lib/algebra/set/fixnum_spec.rb
	spec/lib/algebra/set/float_spec.rb
	spec/lib/algebra/set/log_spec.rb
	spec/lib/algebra/set/num_spec.rb
	spec/lib/algebra/set/pi_spec.rb
	spec/lib/algebra/set/sin_spec.rb
	spec/lib/algebra/set/symbol_spec.rb
	spec/lib/algebra/set/tan_spec.rb
	spec/lib/delta_spec.rb
	spec/lib/function_spec.rb
	spec/lib/integrand_spec.rb

huyu398 committed:
Improve rubocop rules on .rubocop.yml.

huyu398 committed:
Remodify `dydx` spec with rubocop rules.

huyu398 committed:
Reimprove rubocop rules on .rubocop.yml.

huyu398 committed:
Remodify `dydx` lib with rubocop rules.

huyu398 committed:
Add `rubocop` gem to Gemfile.

huyu398 commented:
@gogotanaka 
rspec のコード修正を行い， merge + solve to conflict 後， 再度 lib, spec ともにコード修正を行いました．
コード修正は簡単な caution, warning のみに留めました．いくつかの caution, warning は簡単に修正できますが，別ブランチに分けたほうが良いと考えたため，ここには含めていません（ `is_hoge?` メソッド類）．

確認お願いします :)

gogotanaka commented:
@huyu398 すばらしいです。。。。。！！！！ありがとうございます！

gogotanaka commented on spec/dydx_spec.rb:
```diff
@@ -6,15 +6,15 @@
     expect(f(1, a + b)).to eq(1 + 2 * ( a + b ))
     expect(f(1, 2)).to eq(5)
-    expect(d/dx(f(x, y))).to eq(1 + y)
+    expect(d / dx(f(x, y))).to eq(1 + y)
```
`d/dx` こちらに関してはスペースがない方が良い気もしますね.

huyu398 replied:
最初は rubocop の caution を回避するために変更していました．
しかし， `**` 演算子の caution （ `**` の左右にスペースは不要）を無視するため `.rubocop.yml` に `SpaceArroundOperators:` を加えたところ， `/` 演算子の左右にスペースがなくても caution 回避できるようになりました．
こうなると， `d/dx` のほうがよさそうですね．

huyu398 committed:
Restore `/` around spaces.

Restored spaces are only differential operator like `dydx`.
Division operator aren't restored.

huyu398 commented:
@gogotanaka 

``` sh
$ find spec | xargs grep ' / d'
```

で検出されるすべての `/` に対して，周囲のスペースを除きました．
通常の割り算の `/` に対しては，周囲のスペースを残したままです．

これでおそらく `d / dx` のような部分は， `d/dx` になっているかと思います．

gogotanaka commented:
@huyu398 問題なさそうです.

すばらしい試みだと思います :sparkles:

huyu398 commented:
@gogotanaka 
merge されずに close されたっぽいですが，大丈夫なのでしょうか？

gogotanaka commented:
@huyu398 うわ。。。。これは誠に申し訳ありません不注意でした。。。。

本当にすみません。

huyu398 commented:
@gogotanaka 
いえいえ，お気になさらず :)
